.github/workflows/ci.yml

```diff
@@ -35,7 +35,7 @@ jobs:
             asan: OFF
             regression-tests: true
 
-          - llvm: 14
+          - llvm: 17
```

Comment:
Is this change necessary? I thought that there was a reason why we run the address-sanitization CI on LLVM 14, that's why I didn't change this in the LLVM 16->17 update PR (I assumed this was the case because otherwise it would have been `llvm: 16` at the time but maybe we just forgot to update it somewhere along the way)

Reply:
> Is this change necessary?

Unfortunately, it is. As I described in the second commit message, it was necessary to add the `DIBuilder::createFunction` method to the tests so it works correctly. The method (respectively the `DISubprogram` object) looks like was leaking until LLVM 17 - https://www.github.com/llvm/llvm-project/commit/9eda46b1817f60378143846477ef71f0131a121d, so the tests would not pass for LLVM 14.

> I thought that there was a reason why we run the address-sanitization CI on LLVM 14

I hope there is not a reason, it looks like that the run was added ([commit](https://github.com/diffkemp/diffkemp/commit/e02cd02faa20efde3042eb73fa982a54443497a0) ) when LLVM 14 was the latest LLVM that we used in the CI. So probably we just forgot to update it with the following LLVM releases.

Reply:
I looked at PRs connected to changes in the workflow:
- PR for LLVM 15 (#242) was opened on Aug 18, 2022 and merged on Nov 2, 2022
- PR for ASAN (#243) was opened on Oct 3, 2022 and was merged on Oct 19, 2022

So the ASAN PR was opened after LLVM 15 PR but before it was merged. When the ASAN was merged, the latest supported LLVM was still LLVM 14, the LLVM 15 PR was still being reviewed. So I think that the reason why the ASAN is run on LLVM 14 is that before LLVM 15 was merged it was rebased on the master containing the ASAN but the version on which the ASAN is run was forgotten to be updated. Just to be sure, @zacikpa do you know if there is some specific reason why ASAN is run on LLVM 14 or can it be updated?

Reply:
I'm pretty sure LLVM 14 was picked because it was the latest one supported at the time.